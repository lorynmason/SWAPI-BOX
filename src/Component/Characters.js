import React from 'react'
import './styles/main.scss';
import PropTypes from 'prop-types'

const Characters = ({characters}) => {
  const cards = characters.map( character => {
    return (<div className="character-card card">
      <h1>{character.name}</h1>
      <p> Species: {character.species}</p>
      <p> Homeworld: {character.homeworld}</p>
      <p> Population: {character.population}</p>
   </div>)
  })
 
  return (
  <section className="cardContainer">
    {cards}
  </section>
  )
}

Characters.propTypes = {
  
}

export default Characters