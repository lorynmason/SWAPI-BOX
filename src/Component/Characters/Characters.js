import React from 'react'
import '../styles/main.scss'
import PropTypes from 'prop-types'

const Characters = ({ characters, toggleFavorites }) => {
  const cards = characters.map((character) => {
    const infoValue = Object.values(character.info).map((chInfo) => {
      return (
        <p key={chInfo[0]}>{chInfo}</p>
      )
    })
    return (
      <div className="character-card card" key={character.id}>
        <button className="favorite-btn" type="button" onClick={() => toggleFavorites(character.id)}>
          <i className="fas fa-jedi" />
        </button>
        <div className="info">
          <p>
            {infoValue}
          </p>
        </div>
      </div>
    )
  })
  return (
    <section className="cardContainer">
      {cards}
    </section>
  )
}

Characters.propTypes = {
  characters: PropTypes.array.isRequired,
  toggleFavorites: PropTypes.func.isRequired
}

export default Characters
