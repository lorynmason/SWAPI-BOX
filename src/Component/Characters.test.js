import React from 'react'
import Characters from './Characters'
import { shallow } from 'enzyme'

describe('Characters', () => {
  let wrapper
  let mockCharacters
  let mockActivePage
  let mockToggleFavorites
  let mockFavorites
  let mockCardId

  beforeEach(() => {
    mockFavorites = []
    mockCharacters = [
        {
          id: 'bob',
          category: 'characters',
          info: {Homeworld: 'Homeworld:  Earth',
            Population: 'Population:  123456789',
            Species: 'Species:  human',
            Name: 'Name:  bob'}
        }
      ]
    mockActivePage = 'mockCharacters'
    mockToggleFavorites = jest.fn(mockCharacters.id)
    wrapper = shallow(<Characters characters={mockCharacters} toggleFavorites={mockToggleFavorites} activePage={mockActivePage} favorites={mockFavorites}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call toggleFavorites with the correct params', () =>{
    wrapper.find(".favorite-btn").simulate('click')
    expect(mockToggleFavorites).toHaveBeenCalledWith(mockCharacters[0].id)
  })

})