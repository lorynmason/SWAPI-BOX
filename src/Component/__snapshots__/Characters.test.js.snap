// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Characters
    activePage="mockCharacters"
    characters={
      Array [
        Object {
          "category": "characters",
          "id": "bob",
          "info": Object {
            "Homeworld": "Homeworld:  Earth",
            "Name": "Name:  bob",
            "Population": "Population:  123456789",
            "Species": "Species:  human",
          },
        },
      ]
    }
    favorites={Array []}
    toggleFavorites={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="character-card card"
        >
          <button
            className="favorite-btn"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-jedi"
            />
          </button>
          <div
            className="info"
          >
            <p>
              <p>
                Homeworld:  Earth
              </p>
              <p>
                Population:  123456789
              </p>
              <p>
                Species:  human
              </p>
              <p>
                Name:  bob
              </p>
            </p>
          </div>
        </div>,
      ],
      "className": "cardContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "bob",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="favorite-btn"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-jedi"
              />
            </button>,
            <div
              className="info"
            >
              <p>
                <p>
                  Homeworld:  Earth
                </p>
                <p>
                  Population:  123456789
                </p>
                <p>
                  Species:  human
                </p>
                <p>
                  Name:  bob
                </p>
              </p>
            </div>,
          ],
          "className": "character-card card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-jedi"
              />,
              "className": "favorite-btn",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-jedi",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                <p>
                  Homeworld:  Earth
                </p>
                <p>
                  Population:  123456789
                </p>
                <p>
                  Species:  human
                </p>
                <p>
                  Name:  bob
                </p>
              </p>,
              "className": "info",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Homeworld:  Earth
                  </p>,
                  <p>
                    Population:  123456789
                  </p>,
                  <p>
                    Species:  human
                  </p>,
                  <p>
                    Name:  bob
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "H",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Homeworld:  Earth",
                  },
                  "ref": null,
                  "rendered": "Homeworld:  Earth",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": "P",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Population:  123456789",
                  },
                  "ref": null,
                  "rendered": "Population:  123456789",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": "S",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Species:  human",
                  },
                  "ref": null,
                  "rendered": "Species:  human",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": "N",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Name:  bob",
                  },
                  "ref": null,
                  "rendered": "Name:  bob",
                  "type": "p",
                },
              ],
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="character-card card"
          >
            <button
              className="favorite-btn"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-jedi"
              />
            </button>
            <div
              className="info"
            >
              <p>
                <p>
                  Homeworld:  Earth
                </p>
                <p>
                  Population:  123456789
                </p>
                <p>
                  Species:  human
                </p>
                <p>
                  Name:  bob
                </p>
              </p>
            </div>
          </div>,
        ],
        "className": "cardContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "bob",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="favorite-btn"
                onClick={[Function]}
                type="button"
              >
                <i
                  className="fas fa-jedi"
                />
              </button>,
              <div
                className="info"
              >
                <p>
                  <p>
                    Homeworld:  Earth
                  </p>
                  <p>
                    Population:  123456789
                  </p>
                  <p>
                    Species:  human
                  </p>
                  <p>
                    Name:  bob
                  </p>
                </p>
              </div>,
            ],
            "className": "character-card card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-jedi"
                />,
                "className": "favorite-btn",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-jedi",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  <p>
                    Homeworld:  Earth
                  </p>
                  <p>
                    Population:  123456789
                  </p>
                  <p>
                    Species:  human
                  </p>
                  <p>
                    Name:  bob
                  </p>
                </p>,
                "className": "info",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      Homeworld:  Earth
                    </p>,
                    <p>
                      Population:  123456789
                    </p>,
                    <p>
                      Species:  human
                    </p>,
                    <p>
                      Name:  bob
                    </p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "H",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Homeworld:  Earth",
                    },
                    "ref": null,
                    "rendered": "Homeworld:  Earth",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": "P",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Population:  123456789",
                    },
                    "ref": null,
                    "rendered": "Population:  123456789",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": "S",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Species:  human",
                    },
                    "ref": null,
                    "rendered": "Species:  human",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": "N",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Name:  bob",
                    },
                    "ref": null,
                    "rendered": "Name:  bob",
                    "type": "p",
                  },
                ],
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
